python/lsst/sims/photUtils/SignalToNoise.py

```diff
@@ -0,0 +1,552 @@
+    """
+    Calculate the effective number of pixels in a double Gaussian PSF
+
+    @param [in] seeing in arcseconds
+
+    @param [in] platescale in arcseconds per pixel
+
+    @param [out] the effective number of pixels contained by a double
+    Gaussian PSF
+
+    see equation 31 of the SNR document
+    https://docushare.lsstcorp.org/docushare/dsweb/ImageStoreViewer/LSE-40
+    """
+    return 2.436*(seeing/platescale)**2
+
+
+def calcInstrNoiseSq(photParams):
+    """
+    Combine all of the noise due to intrumentation into one value
+
+    @param [in] photParams is an instantiation of the
+    PhotometricParameters class that carries details about the
+    photometric response of the telescope.
+
+    @param [out] The noise due to all of these sources added in quadrature
+    in ADU counts
+    """
+
+    return (photParams.nexp*photParams.readnoise**2 + \
+           photParams.darkcurrent*photParams.exptime*photParams.nexp + \
+           photParams.nexp*photParams.othernoise**2)/(photParams.gain*photParams.gain)
+
+
+
+def calcTotalNonSourceNoiseSq(skySed, hardwarebandpass, photParams, seeing):
+    """
+    Calculate the noise due to things that are not the source being observed
+    (i.e. intrumentation and sky background)
+
+    @param [in] skySed -- an instantiation of the Sed class representing the sky
+    (normalized so that skySed.calcMag() gives the sky brightness in magnitudes
+    per square arcsecond)
+
+    @param [in] hardwarebandpass -- an instantiation of the Bandpass class representing
+    just the instrumentation throughputs
+
+    @param [in] photParams is an instantiation of the
+    PhotometricParameters class that carries details about the
+    photometric response of the telescope.
+
+    @param [in] seeing in arcseconds
+
+    @param [out] total non-source noise squared (in ADU counts)
+    (this is simga^2_tot * neff in equation 41 of the SNR document
+    https://docushare.lsstcorp.org/docushare/dsweb/ImageStoreViewer/LSE-40 )
+
+    @param [out] noise squared due just to the instrument (in ADU counts)
+
+    @param [out] noise squared due to the sky (in ADU counts)
+
+    @param [out] noise squared due to sky measurement (in ADU counts, presently set to zero)
+
+    @param [out] the effective number of pixels in a double Gaussian PSF
+    """
+
+    # This method outputs all of the parameters calculated along the way
+    # so that the verbose version of calcSNR_sed still works
+
+    # Calculate the effective number of pixels for double-Gaussian PSF
+    neff = calcNeff(seeing, photParams.platescale)
+
+    # Calculate the counts from the sky.
+    # We multiply by two factors of the platescale because we expect the
+    # skySed to be normalized such that calcADU gives counts per
+    # square arc second, and we need to convert to counts per pixel.
+
+    skycounts = skySed.calcADU(hardwarebandpass, photParams=photParams) \
+                * photParams.platescale * photParams.platescale
```

Comment:
Do you want to modify this and the call for this method, so that there is a new method 
calcSkyCounts  (which takes the sed, hardware, and total bandpass and photometric parameters and returns sky counts per pixel)
and then this method would just take the sky counts per pixel? 

It might make it easier to use for people who don't have a sky sed but are calculating calcSkyCountsPerPixelForM5?